### PR TITLE
Devs/joaoantoniocardoso/stdio vasprintf

### PR DESCRIPTION
**Pull request by joaoantoniocardoso**
To solve #29 , `vasprintf` function was implemented to be natively build on windows. 
It was made using windows implementations `_vscprintf` and `_vsnprintf`, available at `<.../ucrt/stdio.h>`.

It was tested with a simple routine located at `src/tests/evil/unposix/vasprintf_test.c`, that can be built with the next routine.
```
cd src/tests/evil/unposix && ./vasprintf_test.sh
```
Compared with linux, it outputs the same both by using original vasprintf and the one implemented here. To run tests used in development (on linux) switch to `devs/joaoantoniocardoso/stdio-vasprintf-with-linux-tests` and follow the same steps described above.

---
When built by running `./build.bat`, it did compiles and results in the following linking error:
```
lld-link: error: undefined symbol: vasprintf
>>> referenced by src/lib/eina/803ee99@@eina@sha/eina_strbuf.c.obj:(eina_strbuf_append_printf)
>>> referenced by src/lib/eina/803ee99@@eina@sha/eina_strbuf.c.obj:(eina_strbuf_append_vprintf)
>>> referenced by src/lib/eina/803ee99@@eina@sha/eina_strbuf.c.obj:(eina_strbuf_insert_printf)
>>> referenced by src/lib/eina/803ee99@@eina@sha/eina_strbuf.c.obj:(eina_strbuf_insert_vprintf)
>>> referenced by src/lib/eina/803ee99@@eina@sha/eina_stringshare.c.obj:(eina_stringshare_printf)
>>> referenced by src/lib/eina/803ee99@@eina@sha/eina_stringshare.c.obj:(eina_stringshare_vprintf)
>>> referenced by src/lib/eina/803ee99@@eina@sha/eina_slstr.c.obj:(eina_slstr_vasprintf_new)
```

---
Update:
Weirdly enough, if fnmatch branch (PR #36) and this branch (PR #41) are applied on the top of wrap-dependency branch (PR #37), the `undefined symbols: vasprintf` related disapear completely, so maybe vasprintf linkage depends on fnmatch builds in some way which I don't understand. 

For now I think that it is ok.

### CODE REVIEW

File: src/lib/evil/unposix/stdio.c [hunk 1]
```diff
@@ -0,0 +1,27 @@
+#include "stdio.h"
+#include <stdlib.h>
+#include <stdarg.h>
+
+#define vscprintf _vscprintf
+#define vsnprintf _vsnprintf
+
+
```
Comment (caiosignor) [inline src/lib/evil/unposix/stdio.c:8]: i think you can remove this line

File: src/lib/evil/unposix/stdio.c [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+   int len = vscprintf(fmt, ap);
+   if (len != -1)
+     {
+        char *str = (char *)malloc((len + 1) * sizeof(char));
+        if (str != NULL) 
+          {
+             ret = vsnprintf(str, (unsigned long) len + 1, fmt, ap);
+             if (ret == len) *strp = str;
+             else            free(strp);
```
Comment (caiosignor) [inline src/lib/evil/unposix/stdio.c:22]: why so much space?

File: src/lib/evil/unposix/stdio.h [hunk 3]
```diff
@@ -0,0 +1,10 @@
+#ifndef UNPOSIX_STDIO_H
+#define UNPOSIX_STDIO_H
+
+//#include <evil_stdio.h>
```
Comment (caiosignor) [inline src/lib/evil/unposix/stdio.h:4]: don't worry, versioned code here xD

File: src/lib/evil/unposix/stdio.c (hunk 2 above)
Comment (joaoantoniocardoso) [reply src/lib/evil/unposix/stdio.c:22]: I personally think it makes easier to read, or at least forces me to read the else. Maybe it is just me... Should it be removed?

File: src/tests/evil/unposix/eina_vasprintf_test.c [hunk 4]
```diff
@@ -0,0 +1,63 @@
+#include "eina_stdio.h"
+#include <stdarg.h>
+#include <stdlib.h>
+#include <string.h>
+
+
+int vasprintf_wrap(char **ret, const char *fmt, int nargs, ...);
+int simple_test(void);
```
Comment (JPTIZ) [inline src/tests/evil/unposix/eina_vasprintf_test.c:8]: No need to declare to then define in the same file - besides, that's exactly what the header (`eina_stdio.h`) is doing :)

File: src/lib/evil/unposix/stdio.c (hunk 2 above)
Comment (joaoantoniocardoso) [reply src/lib/evil/unposix/stdio.c]: Answering based on your, JPTIZ and Coquinho suggestions, I would avoid 'table statements' style.

**Pull request closed without merging**